host-controller/src/main/java/org/jboss/as/host/controller/model/jvm/JvmOptionsBuilderFactory.java

```diff
@@ -24,186 +24,124 @@
+                    continue;
+                }
+                if (!checkOption(maxHeap != null && option.startsWith("-Xmx"), jvmName, option, Element.HEAP.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.getStack() != null && option.startsWith("-Xss"), jvmName, option, Element.STACK.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.isDebugEnabled() != null && jvmElement.isDebugEnabled() && jvmElement.getDebugOptions() != null &&
+                                (option.startsWith("-Xrunjdwp") || option.startsWith("-agentlib:jdwp")),
+                        jvmName, option, Attribute.DEBUG_OPTIONS.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.getAgentPath() != null && option.startsWith("-agentpath:"), jvmName, option, Element.AGENT_PATH.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.getAgentLib() != null && option.startsWith("-agentlib:"), jvmName, option, Element.AGENT_LIB.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.getAgentLib() != null && option.startsWith("-javaagent:"), jvmName, option, Element.AGENT_LIB.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.getJavaagent() != null && option.startsWith("-Xmx"), jvmName, option, Element.JAVA_AGENT.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.getJavaagent() != null && option.startsWith("-XX:PermSize"), jvmName, option, Element.PERMGEN.toString())) {
+                    continue;
+                }
+                if (!checkOption(jvmElement.getJavaagent() != null && option.startsWith("-XX:MaxPermSize"), jvmName, option, Element.PERMGEN.toString())) {
+                    continue;
+                }
+                if (checkAdditionalJvmOption(option)) {
+                    command.add(option);
+                }
             }
-            return true;
         }
-// This builder doesn't override anything currently, since the addPermgen behavior
-        // became standard. But we keep the concept in case we do vm-specific stuff
-        // in the future
     }
 
-    private static class IbmJvmOptionsBuilder extends JvmOptionsBuilder {
-        private IbmJvmOptionsBuilder(JvmType type) {
-            super(type);
+    boolean checkOption(boolean condition, String jvm, String option, String schemaElement) {
+        if (condition) {
+            ROOT_LOGGER.optionAlreadySet(option, jvm, schemaElement);
+            return false;
         }
+        return true;
+    }
 
-        // This builder doesn't override anything currently, since the addPermgen behavior
-        // became standard. But we keep the concept in case we do vm-specific stuff
```

Comment:
@bstansberry @jmesnil Are you ok with removing of this concept?

Reply:
The AddPermGen mechanism was doing nothing so I removed it.

Reply:
Fine with me. It's an internal implementation detail and the implementation long ago drifted away from the original need for IbmJvmOptionsBuilder.